fiftyone/operators/input_utils.py

```diff
@@ -0,0 +1,244 @@
+            view
+        default_target (None): the default target view to select if multiple
+            choices are available. If ``None``, one will be chosen based on the
+            available choices in the following order of preference:
+            dataset view, current view, selected samples, selected labels
+        dataset_label (Entire dataset): the label for the "entire dataset"
+            target view
+        dataset_description (None): the description for the "entire dataset" target
+            view. If ``None``, a default description is generated
+        base_view_label (Base view): the label for the "base view" target view
+        base_view_description (None): the description for the "base view" target
+            view. If ``None``, a default description is generated
+        current_view_label (Current view): the label for the "current view"
+            target view
+        current_view_description (None): the description for the "current view"
+            target view. If ``None``, a default description is generated
+        dataset_view_label (Dataset): the label for the "dataset view" target
+            view
+        dataset_view_description (None): the description for the "dataset view"
+            target view. If ``None``, a default description is generated
+        selected_samples_label (Selected samples): the label for the "selected
+            samples" target view
+        selected_samples_description (None): the description for the "selected
+            samples" target view. If ``None``, a default description is generated
+        selected_labels_label (Selected labels): the label for the "selected
+            labels" target view
+        selected_labels_description (None): the description for the "selected
+            labels" target view. If ``None``, a default description is generated
+
+    Returns:
+        the resolved target view, one of
+        :class:`fiftyone.operators.constants.ViewTarget`
+    """
+    # Resolve descriptions for the various target views
+    action_description = action_description or "Process"
+    dataset_description = (
+        dataset_description or f"{action_description} the entire dataset"
+    )
+    base_view_description = (
+        base_view_description or f"{action_description} the base view"
+    )
+    current_view_description = (
+        current_view_description or f"{action_description} the current view"
+    )
+    dataset_view_description = (
+        dataset_view_description or f"{action_description} the dataset view"
+    )
+    selected_samples_description = selected_samples_description or (
+        f"{action_description} only the selected samples"
+    )
+    selected_labels_description = selected_labels_description or (
+        f"{action_description} only the selected labels"
+    )
+
+    # Determine which target views are available
+    has_base_view = isinstance(ctx.view, fop.PatchesView)
```

Comment:
Since this is a general util function, do frames and clips views also need to be handled?